lighthouse-core/test/runner-test.js

```diff
@@ -126,6 +127,63 @@ describe('Runner', () => {
       });
   });
 
+  it('rejects when domain name can\'t be resolved', () => {
+    const url = 'https://some-non-existing-domain.com';
+    const config = new Config({
+      passes: [{
+        gatherers: ['https']
+      }],
+      audits: [
+        'is-on-https'
+      ]
+    });
+
+    // Arrange for driver to return unresolved request.
+    const unresolvedDriver = Object.assign({}, driverMock, {
+      online: true,
+      endNetworkCollect() {
+        return Promise.resolve(
+          recordsFromLogs(require('./fixtures/unresolved-perflog.json'))
+        );
+      }
+    });
+
+    return Runner.run({}, {url, config, driverMock: unresolvedDriver})
+      .then(_ => {
+        assert.ok(false);
+      }, error => {
+        assert.ok(true);
```

Comment:
what do you mean? The result throws when `run` resolves and resolves when `run` throws. What do you suggest instead?